### PR TITLE
Added coinview test

### PR DESCRIPTION
**Pull request by noescape00**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus.Tests/CoinViews/CoinviewTests.cs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            this.rewindDataIndexCache.InitializeAsync(this.concurrentChain.Height, this.cachedCoinView);
+
+            this.random = new Random();
+
+            ChainedHeader newTip = ChainedHeadersHelper.CreateConsecutiveHeaders(1000, this.concurrentChain.Tip, true, null, this.network).Last();
+            this.concurrentChain.SetTip(newTip);
+        }
+
+        [Fact]
+        public async Task TestRewindAsync()
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.Consensus.Tests/CoinViews/CoinviewTests.cs]: This tests only works on the cached data right? (I did not notice a call to flush while you aggregate)

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus.Tests/CoinViews/CoinviewTests.cs]: We should probably add to the test a mixture of flush and none flush cases, should be fairly easy no?

I suspect the bip30 bug happened after a double reorg meaning we cleared the cache on the first reorg.
